File: C#Integration/PokerProtocol/Commands/Lobby/IdentifyCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby
{
    public class IdentifyCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return IdentifyCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyIDENTIFY";

        private readonly string m_Name;


        public string Name
        {
            get { return m_Name; }
        } 


        public IdentifyCommand(StringTokenizer argsToken)
        {
            m_Name = argsToken.NextToken();
        }

        public IdentifyCommand(string name)
        {
            m_Name = name;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_Name);
        }

        public string EncodeResponse( bool success )
        {
            return new IdentifyResponse(this, success).Encode();
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/protocol/commands/lobby/IdentifyCommand.java
package protocol.commands.lobby;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;
import protocol.commands.lobby.response.IdentifyResponse;

public class IdentifyCommand implements ICommand
{
    public static String COMMAND_NAME = "lobbyIDENTIFY";
    
    private final String m_playerName;
    
    public IdentifyCommand(StringTokenizer argsToken)
    {
        m_playerName = argsToken.nextToken();
    }
    
    public IdentifyCommand(String name)
    {
        m_playerName = name;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(IdentifyCommand.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        sb.append(m_playerName);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String encodeResponse(Boolean success)
    {
        return new IdentifyResponse(m_playerName, success).encodeCommand();
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
}
